boards/shields/x_nucleo_53l0a1/Kconfig.defconfig

```diff
@@ -0,0 +1,24 @@
+# Copyright (c) 2021 Titouan Christophe
+# SPDX-License-Identifier: Apache-2.0
+
+if SHIELD_X_NUCLEO_53L0A1
+
+config GPIO
+	default y
```

Comment:
@erwango after removing the above, this fails in buildkite: https://buildkite.com/zephyr/zephyr/builds/40215#43279ba5-fbc3-45aa-a3d8-bbd7383dd969

I can reproduce on my laptop:
```
$ west build -b sam_v71_xult zephyr/samples/shields/x_nucleo_53l0a1/
[...]
/opt/zephyr-sdk-13/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/10.3.0/../../../../arm-zephyr-eabi/bin/ld.bfd: app/libapp.a(main.c.obj): in function `main':
/home/titou/Code/midizephyr/zephyr/samples/shields/x_nucleo_53l0a1/src/main.c:149: undefined reference to `__device_dts_ord_14'
/opt/zephyr-sdk-13/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/10.3.0/../../../../arm-zephyr-eabi/bin/ld.bfd: zephyr/drivers/sensor/vl53l0x/libdrivers__sensor__vl53l0x.a(vl53l0x.c.obj):(.data.vl53l0x_0_driver+0x8): undefined reference to `__device_dts_ord_105'
/opt/zephyr-sdk-13/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/10.3.0/../../../../arm-zephyr-eabi/bin/ld.bfd: zephyr/drivers/sensor/vl53l0x/libdrivers__sensor__vl53l0x.a(vl53l0x.c.obj):(.data.vl53l0x_1_driver+0x8): undefined reference to `__device_dts_ord_107'
/opt/zephyr-sdk-13/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/10.3.0/../../../../arm-zephyr-eabi/bin/ld.bfd: zephyr/drivers/sensor/vl53l0x/libdrivers__sensor__vl53l0x.a(vl53l0x.c.obj):(.data.vl53l0x_2_driver+0x8): undefined reference to `__device_dts_ord_107'
collect2: error: ld returned 1 exit status
```

This is caused by the references like `<&expander1 15 0>`, from the expansion board overlay, not being found. And indeed:
```
$ grep GPIO build/zephyr/.config
# CONFIG_GPIO is not set
CONFIG_GPIO_STMPE1600=y
# CONFIG_I2C_GPIO is not set
```

The GPIO expanders are never instanciated, because the GPIO subsystem is disabled, hence the missing references.

Therefore, I believe that adding this default is required. What do you think ?